src/mimir_coordinator.py

```diff
@@ -163,6 +182,20 @@ def _build_compactor_config(self) -> Dict[str, Any]:
             "data_dir": str(self._root_data_dir / "data-compactor"),
         }
 
+    # ring.replication_factor: int
+    # Number of ingesters that each time series is replicated to. This option
+    # needs be set on ingesters, distributors, queriers and rulers when running in
+    # microservices mode.
+    def _build_ingester_config(self) -> Dict[str, Any]:
+        ingester_scale = len(self._cluster_provider.gather_addresses_by_role()[MimirRole.ingester])
```

Comment:
The same may occur here.

```suggestion
        ingester_scale = len(self._cluster_provider.gather_addresses_by_role().get(MimirRole.ingester, None))
```